python/perf-kernels/fused_moe/moe-gemm.py

```diff
@@ -299,13 +380,54 @@ def moe_gemm(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, topk_weights: to
 
     EVEN_K = K % config["BLOCK_SIZE_K"] == 0
 
-    moe_gemm_kernel[grid](a, b, c, a.stride(0), a.stride(1), b.stride(0), b.stride(1), b.stride(2), c.stride(1),
-                          c.stride(2), top_k, topk_weights, sorted_token_ids, expert_ids, EM, N, K, EVEN_K,
-                          MUL_ROUTED_WEIGHT=topk_weights is not None, **config)
+    moe_gemm_kernel[grid](a, b, c, a_descale,
+                          b_descale, a.stride(0), a.stride(1), b.stride(0), b.stride(1), b.stride(2), c.stride(1),
+                          c.stride(2), stride_bse, stride_bsn, top_k, topk_weights, sorted_token_ids, expert_ids, EM, N,
+                          K, EVEN_K, MUL_ROUTED_WEIGHT=topk_weights is not None, use_fp8_w8a8=use_fp8_w8a8,
+                          use_int8_w8a16=use_int8_w8a16, **config)
     return c
 
 
-def input_helper(M: int, N: int, K: int, top_k: int, E: int, routed_weight: bool, dtype):
+def quantize_tensor(tensor: torch.Tensor, dtype, dim=()) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    quantize_dim = [i for i in range(tensor.dim()) if i not in dim]
+    max_vals = tensor.abs().amax(dim=quantize_dim, keepdim=True)
+    max_repr_val = dtype_max[dtype]
+    # Avoid division by zero
+    max_vals[max_vals == 0] = 1e-8
+
+    # Compute scale factors for each channel
+    scale: torch.Tensor = max_repr_val / max_vals.to(torch.float32)
```

Comment:
So for a tensor of shape M x K, what is the shape of the scale?

Reply:
The tensor shapes
```
    a = torch.randn((M, K), dtype=dtype, device='cuda')
    b = torch.randn((E, N, K), dtype=dtype, device='cuda')
```

In the case of fp8_w8a8:
- a_descale is a scalar
- b_descale is (E, ), per expert 

in the case of use_int8_w8a16:
- b_quantized is (E, N), per expert and per n